driver_with_mockserver_test.go

```diff
@@ -1704,7 +1704,160 @@ func TestTransactionBatchDml(t *testing.T) {
 	if len(commitRequests) != 1 {
 		t.Fatalf("Commit requests count mismatch\nGot: %v\nWant: %v", len(commitRequests), 1)
 	}
+}
+
+func TestCommitTimestamp(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	db, _, teardown := setupTestDBConnection(t)
+	defer teardown()
+
+	conn, err := db.Conn(ctx)
+	defer conn.Close()
+	if err != nil {
+		t.Fatalf("failed to get a connection: %v", err)
+	}
+	tx, err := conn.BeginTx(ctx, nil)
+	if err != nil {
+		t.Fatalf("failed to start transaction: %v", err)
+	}
+	if err := tx.Commit(); err != nil {
+		t.Fatalf("commit failed: %v", err)
+	}
+	// Get the commit timestamp from the connection.
+	// We do this in a simple loop to verify that we can get it multiple times.
+	for i := 0; i < 2; i++ {
+		var ts time.Time
+		if err := conn.Raw(func(driverConn interface{}) error {
+			ts, err = driverConn.(SpannerConn).CommitTimestamp()
+			return err
+		}); err != nil {
+			t.Fatalf("failed to get commit timestamp: %v", err)
+		}
+		if cmp.Equal(time.Time{}, ts) {
+			t.Fatalf("got zero commit timestamp: %v", ts)
+		}
+	}
+}
+
+func TestCommitTimestampAutocommit(t *testing.T) {
+	t.Parallel()
 
+	ctx := context.Background()
+	db, _, teardown := setupTestDBConnection(t)
+	defer teardown()
+
+	conn, err := db.Conn(ctx)
+	defer conn.Close()
+	if err != nil {
+		t.Fatalf("failed to get a connection: %v", err)
+	}
+	if _, err := conn.ExecContext(ctx, testutil.UpdateBarSetFoo); err != nil {
```

Comment:
this will do an autocommit, right?

Reply:
Yes, that is correct. If you execute an update (or any other statement) on a connection that has no active transaction, the statement will be executed in autocommit mode. That means that:
1. Update statements will automatically begin an implicit read/write transaction and commit that transaction directly after the update statement if the update was successful.
2. Queries will automatically use a single-use read-only transaction, as that is the most efficient way of executing a single query.